src/python/gem5/resources/resource.py

```diff
@@ -554,6 +565,153 @@ def _get_weights_and_simpoints_from_file(
+            description=description,
+            source=source,
+            resource_version=resource_version,
+        )
+
+    def __iter__(self) -> Generator["WorkloadResource", None, None]:
+        """
+        Returns a generator that iterates over the workloads in the suite.
+
+        :returns: A generator that iterates over the workloads in the suite.
+        """
+        for workload in self._workload_obj_list:
+            yield workload
+
+    def __len__(self):
+        """
+        Returns the number of workloads in the suite.
+
+        :returns: The number of workloads in the suite.
+        """
+        return len(self._workload_obj_list)
+
+    def with_input_group(self, input_group: str) -> "SuiteResource":
+        """
+        Returns a new SuiteResource object with only the workloads that use the
+        specified input group.
+
+        :param input_group: The input group to filter the workloads by.
+        :returns: A new SuiteResource object with only the workloads that use
+        the specified input group.
+        """
+        if self._workloads == None:
+            warn(
+                "Optional parameter 'workloads' which contains resource id, version,"
+                " and input group not found."
+            )
+            return SuiteResource(
+                local_path=self._local_path,
+                resource_version=self._resource_version,
+                workloads=self._workloads,
+                description=self._description,
+                source=self._source,
+                workload_obj_list=[],
+            )
+
+        filtered_workloads = []
+        filtered_workload_obj_list = []
+        for i in range(len(self._workloads)):
+            if input_group in self._workloads[i]["input_group"]:
+                filtered_workloads.append(self._workloads[i])
+                filtered_workload_obj_list.append(self._workload_obj_list[i])
+
+        return SuiteResource(
+            local_path=self._local_path,
+            resource_version=self._resource_version,
+            workloads=filtered_workloads,
+            description=self._description,
+            source=self._source,
+            workload_obj_list=filtered_workload_obj_list,
+        )
+
+    def get_input_groups(self) -> Set[str]:
+        """
+        Returns a set of all input groups used by the workloads in a suite.
+
+        :returns: A set of all input groups used by the workloads in a suite.
+        """
+        if self._workloads == None:
+            warn(
+                "Optional parameter 'workloads' which contains resource id, version,"
+                " and input group not found."
+            )
+            return set()
+        return {
+            group
+            for workload in self._workloads
+            for group in workload["input_group"]
+        }
+
+    def get_all_workloads(
+        self,
+    ) -> Generator[Tuple[str, str, List[str], "WorkloadResource"], None, None]:
```

Comment:
This isn't necessary, the class already implements `__iter__` which does exactly this.

Also, `get_all_workloads` sounds like it should return a list or set of workloads, not be a generator (in this case I'd expect it be called something like `next_workload`.)

Reply:
Should we return a list of tuples or just change the function name?

Reply:
I think this function serves a different purpose than `__iter__` since it also returns the id, version, and inputs. 

I also think we're doing a bit of bikeshedding here.

But, to argue about the color of the shed... One option would be to make this the implementation of `__iter__` and to have this function just return the list of workloads.

IMO, the decision here doesn't really matter as long as we pick something that is consistent and we document it.

Reply:
Just get rid of this function. It's weird. If we need the id you can just do `workload.get_id() for workload in suite`. I really don't see any advantage of this function. The workloads which return via the iterator contain all the information you'd ever require.